jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py

```diff
@@ -80,7 +80,11 @@ def place_slit_overlay(self, *args, **kwargs):
 
         # 'S_REGION' contains slit information. Bypass in case no images exist.
         if len(image_data) > 0:
-            if len(spec2d_data) > 0 and 'S_REGION' in spec2d_data[0].meta:
+            # Only use S_REGION for Nirspec data, turn the plugin off
+            # if other data is loaded
+            if len(spec2d_data) > 0 and 'S_REGION' in spec2d_data[0].meta \
+                    and 'INSTRUME' in spec2d_data[0].meta \
+                    and spec2d_data[0].meta['INSTRUME'].lower() == "nirspec":
```

Comment:
Usage of `\` line continuation is discouraged. Also, the logic can be a little shorter?

```suggestion
            # Only use S_REGION for Nirspec data, turn the plugin off
            # if other data is loaded
            if (len(spec2d_data) > 0 and 'S_REGION' in spec2d_data[0].meta
                    and spec2d_data[0].meta.get('INSTRUME', '').lower() == "nirspec"):
```